Tests/Document/DocumentTests.swift

```diff
@@ -0,0 +1,897 @@
+
+    func test_move_elements_at_the_last_of_array() {
+        let target = Document(key: "test-doc")
+        target.update { root in
+            root.data = [Int64(0), Int64(1), Int64(2)]
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,1,2]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 3)
+
+        target.update { root in
+            let data = root.data as? JSONArray
+            let two = data?.getElement(byIndex: 2) as? CRDTElement
+            try? data?.moveLast(id: two!.getID())
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,1,2]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 3)
+
+        target.update { root in
+            let data = root.data as? JSONArray
+            data?.append(Int64(3))
+            let two = data?.getElement(byIndex: 2) as? CRDTElement
+            try? data?.moveLast(id: two!.getID())
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"data":[0,1,3,2]}
+                           """)
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,1,3,2]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 4)
+    }
+
+    func test_simple_move_elements_at_the_last_of_array() {
+        let target = Document(key: "test-doc")
+        target.update { root in
+            root.data = [Int64(0), Int64(1), Int64(2)]
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,1,2]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 3)
+
+        target.update { root in
+            let data = root.data as? JSONArray
+            data?.append(Int64(3))
+            let one = data?.getElement(byIndex: 1) as? CRDTElement
+            try? data?.moveLast(id: one!.getID())
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"data":[0,2,3,1]}
+                           """)
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,2,3,1]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 4)
+    }
+
+    private var cancellables = Set<AnyCancellable>()
+
+    func test_change_paths_test_for_object() {
+        var paths = [String]()
+        let target = Document(key: "test-doc")
+
+        target.eventStream.sink { _ in
+
+        } receiveValue: { event in
+            XCTAssertEqual(event.type, .localChange)
+            XCTAssertEqual((event as? LocalChangeEvent)?.value[0].paths, ["$."])
```

Comment:
I guess that there is an error in tests(```change paths test for object```) of js-sdk.
You can see the log below when running tests of js-sdk.
```
ERROR: 'YORKIE E:', AssertionError{message: 'expected [ '$.', '$.obj' ] to deeply equal [ Array(7) ]', showDiff: true, actual: ['$.', '$.obj'], expected: ['$.', '$.obj', '$.obj.a', '$.obj', '$.obj.\$\.hello', '$.obj', '$'], operator: 'deepStrictEqual’}
```